### PR TITLE
[FEATURE] adds eager `RSVP.reduce`

### PR DESCRIPTION
**Pull request by bekzod**
reused and fixed tests from https://github.com/tildeio/rsvp.js/commit/a00c9ecc3a95cd29afe95ea49c7b31c5a81b5e3b

### CODE REVIEW

**Comment by bekzod** [comment]
this also could be reworked using same approach from #456

File: lib/rsvp/reduce.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      } else {
+        return total;
+      }
+    });
+}
+
+export default function reduce(promises, fn, total, label) {
+  return Promise.all(promises, label)
+    .then(values => {
+      if (!isFunction(fn)) {
```
Comment (stefanpenner) [inline lib/rsvp/reduce.js]: as with your other PR, we should move this to eager.

**Comment by stefanpenner** [comment]
left a quick note, but otherwise this looks great.

**Comment by bekzod** [comment]
reduce is eager now @stefanpenner

**Comment by bekzod** [comment]
@stefanpenner maybe time to merge this :P would be nice feature

**Comment by bekzod** [comment]
ping ping (shall we merge this :P)

**Comment by bekzod** [comment]
would be sweet add this too :P

**Comment by bekzod** [comment]
ping

**Comment by wagenet** [comment]
Closing due to staleness.

**Pull request closed without merging**